### ISSUE TITLE
Server Side emit with Ack not working.

### ISSUE THREAD

**Issue by willjk** [socketio/socket.io#4559]
**Describe the bug**

Client-A emits-with-ack to the Server and server emits-with-ack to Client-B which calls back to Server with data which calls back to Client-A with that data. Client-B properly sends back the data in a callback but the Server always receives null in the Server-callback.

**To Reproduce**

Please fill the following code example:

Socket.IO server version: `4.5.4`

*Server*

```js
import { Server } from "socket.io";

const io = new Server(3000, {});

io.on("connection", (socket) => {
  console.log(`connect ${socket.id}`);

  socket.on("host:broadcast", async (data, callback) => {
		const room = socket.to(data.room);
// I need timeout or the broadcast-operator automatically hits line 233 and "Timeouts"
		room.timeout(5000).emit("member:message",  data, (arg) => {
//arg is always null in 4.5.4
			callback(arg);
		});
	});
});
```

Socket.IO client version: `4.5.4`

*Client*

```js
import { io } from "socket.io-client";

const socket = io("ws://localhost:3000/", {});

socket.on("connect", () => {
  console.log(`connect ${socket.id}`);
                socket.emit("host:broadcast", "data", (res) => {
			console.log(res);
		});
});

socket.on("member:message", (data, cb) => {
			cb({
				success: true,
				payload: "test",
				errorMessage: ""
			});
		});

socket.on("disconnect", () => {
  console.log("disconnect");
});
```

**Expected behavior**

**Platform:**
 - Device: [e.g. Samsung S8]
 - OS: [e.g. Android 9.2]

**Additional context**
I was able to get data back correctly when I went into `broadcast-operator.ts` on line 247 and replace it with this `ack.apply(this, responses);` I know this is probably incorrect but it should point directly to the problem

**Comment by darrachequesne** [comment]
Hi! The first argument is an error object, in case the server does not receive all acknowledgements in time:

```js
io.to("room123").timeout(5000).emit("hello", "world", (err, responses) => {
  if (err) {
    // some clients did not acknowledge the event in the given delay
  } else {
    console.log(responses); // one response per client
  }
});
```

Reference: https://socket.io/docs/v4/broadcasting-events/#with-acknowledgements

**Comment by willjk** [comment]
@darrachequesne  This makes sense. In the documentation for the server the namespace emit shows the [ack with 2 parameters](https://socket.io/docs/v4/server-api/#namespaceemiteventname-args) but on the socket method it [shows it with only 1 parameter](https://socket.io/docs/v4/server-api/#socketemiteventname-args) That's where I was getting mixed up. Thanks for this

**Comment by darrachequesne** [comment]
Yes, the API (with and without error) is indeed a bit misleading, sorry for that. We might add a default timeout in the next major version, to make it clearer.
